### PR TITLE
Activity navigation implementation

### LINKED ISSUE #37: NavActivity base class
Our activities need a way to move their data forward until the final activity, where the booking will be made.

For easing this and making the codebase more legible, a `NavActivity` base class, which will be extended by every other class in the app will provide the following functionality:

- **Method for starting the next activity**, `advanceActivity`: Takes a lambda, which will define the next intent about to be launched.
~- **Method for collecting this activity's extra object parcels**, `collectData`: An abstract method, which will receive the `Intent` created by `advanceActivity`, and will be free to overload it with custom extras (though it's intended to put only the current activity extras).~

Finally, an activity based on the resulting `Intent` will be started.

*Edit*: `collectData` won't be needed anymore, as NavActivity class is able to access the current `Intent` and directly copy the data onto the new `Intent`

### PR DESCRIPTION
**Pull request by alexaib2002**
This activity allows users to traverse all app activities from start to end. This functionality is provided by:

- #37 for moving data between intents
- Several refactors to 
    - #12 , 
    - #13 , 
    - #14 , 
    - #10 , and 
    - #11 

Note the app won't query the DB yet, it will only move users' input up until the summary.

## TODO

- For better development pacing, there are several branches merged here that aren't merged onto `main`, and may be subject to be refactored. 
- This WILL undoubtedly create conflicts with the upstream `main` branch (I will fix them as soon as this branch ends up being ready for review), so don't try merging this branch onto any other locally, until it has been officially merged.

### CODE REVIEW

**Comment by alexaib2002** [comment]
After so much time of hard work, this PR will bring many, many features to the project. And now's ready for review! 

Overall, `main` will be flooded with functional, navigable activities, and all of them will support moving their data up to the last activity. 

Not only that, but users may change their selections dynamically (ie: users may change their seat reservations).

Please, before accepting the changes, checkout this branch on your local repos, and review everything as deeply as you can. As this branch is the merge of lots of commits, it may contain some overlooked files or code (though I did my best to find those).

File: app/CineHub/app/src/main/java/org/cinehub/enums/SeatType.java [hunk 1]
```diff
@@ -0,0 +1,18 @@
+package org.cinehub.enums;
+
+import org.cinehub.R;
+
+public enum SeatType {
```
Comment (Jkutkut) [inline app/CineHub/app/src/main/java/org/cinehub/enums/SeatType.java:5]: Nice use of enum

**Review by Jkutkut — APPROVED** [review]
All looking good to me

**Review by aleexgheorghe — APPROVED** [review]
Apparently it looks like everything is fine! It has worked very well :D

**Pull request merged**